Microsoft.Toolkit.Uwp.UI/AdvancedCollectionView/AdvancedCollectionView.cs

```diff
@@ -555,15 +547,20 @@ private void HandleFilterChanged()
             }
 
             var viewHash = new HashSet<object>(_view);
```

Comment:
This section would probably also benefit from the algorithm at https://github.com/Microsoft/UWPCommunityToolkit/pull/1627#discussion_r151217217
```csharp
for (int index = 0, viewIndex = 0; index < _sourceList.Count; index++)
{
    if (viewIndex == _view.Count || _view[viewIndex] != _sourceList[index])
    {
        if (HandleItemAdded(index, item, viewIndex))
        {
            viewIndex++;
        }
    }
    else
    {
        viewIndex++;
    }
}
```

I think that the current implementation fails for duplicate entries; i.e., a `_sourceList` of `{ "A", "B", "C", "A", "A" }` would only put one `"A"` into the `_view`

Reply:
So, in this case, we we only add in to the view a single item, when we could possibly need to add more than that.  Let me write a test for this behavior as well.

Reply:
So I wrote a some tests and I realize that this is actually not going to cause a problem.  The view hash does not live update with additions so there will never be a case where some items (that are duplicates) are added to the view, and others are not.

I'll add the tests to this PR because I see them as useful for the future! Thanks for the scrutiny! 👍